@navikt/core/react/src/overlays/dismissablelayer/DismissableLayer.tsx

```diff
@@ -331,53 +260,153 @@ const DismissableLayerNode = forwardRef<HTMLDivElement, DismissableLayerProps>(
+        context.branchedLayers.set(parentBranchedLayer, new Set());
+      }
+
+      const branchedChildren = context.branchedLayers.get(parentBranchedLayer)!;
+      branchedChildren.add(node);
+      dispatchUpdate();
+
+      return () => {
+        // Remove this node from the parent's children
+        branchedChildren.delete(node);
+
+        // If the parent has no more children, remove the parent from branchedLayers
+        if (branchedChildren.size === 0) {
+          context.branchedLayers.delete(parentBranchedLayer);
+        }
+
+        dispatchUpdate();
+      };
+    }, [node, parentBranchedLayer, context]);
+
+    /**
+     * Force update when context changes to update index and pointer-events state.
+     * We use a custom event to avoid unnecessary renders from other state changes in the context.
+     */
+    useEffect(() => {
+      const handleUpdate = () => force({});
+      document.addEventListener(CONTEXT_UPDATE_EVENT, handleUpdate);
+      return () =>
+        document.removeEventListener(CONTEXT_UPDATE_EVENT, handleUpdate);
+    }, []);
 
     const Comp = asChild ? Slot : "div";
 
     return (
-      <Comp
-        ref={mergedRefs}
-        {...rest}
-        onFocusCapture={focusOutside.onFocusCapture}
-        onBlurCapture={focusOutside.onBlurCapture}
-        onPointerDownCapture={pointerDownOutside.onPointerDownCapture}
-        style={{
-          pointerEvents: pointerState.pointerStyle,
-          ...rest.style,
-        }}
-      >
-        {children}
-      </Comp>
+      <BranchedLayerContext.Provider value={node}>
+        <Comp
+          {...restProps}
+          ref={mergedRefs}
+          style={{
+            pointerEvents: isBodyPointerEventsDisabled
+              ? isPointerEventsEnabled
+                ? "auto"
+                : "none"
+              : undefined,
+            ...restProps.style,
+          }}
+          onFocusCapture={composeEventHandlers(
+            restProps.onFocusCapture,
+            focusOutside.onFocusCapture,
+          )}
+          onBlurCapture={composeEventHandlers(
+            restProps.onBlurCapture,
+            focusOutside.onBlurCapture,
+          )}
+          onPointerDownCapture={composeEventHandlers(
+            restProps.onPointerDownCapture,
+            pointerDownOutside.onPointerDownCapture,
+          )}
+        >
+          {children}
+        </Comp>
+      </BranchedLayerContext.Provider>
     );
   },
 );
 
+/**
+ * Dispatches a custom event to inform all `DismissableLayer` components to update.
+ */
+function dispatchUpdate() {
+  const event = new CustomEvent(CONTEXT_UPDATE_EVENT);
+  document.dispatchEvent(event);
+}
+
+/**
+ * Returns the highest index of a disabled layer in the ordered layers array.
+ * If no layers are disabled, returns -1.
+ */
+function getHighestDisabledLayerIndex(
+  orderedLayers: DismissableLayerElement[],
+  disabledLayers: Set<DismissableLayerElement>,
```

Comment:
When reading this it was a bit confusing that it only says "disabled layers", when it's actually layers with outside pointer events disabled.

I suggest either including that in the comments and naming:

```suggestion
/**
 * Returns the index of the highest layer with outside pointer events disabled.
 * Returns -1 if no layers are found.
 */
function getHighestLayerWithOutsidePointerEventsDisabledIndex(
  orderedLayers: DismissableLayerElement[],
  layersWithOutsidePointerEventsDisabled: Set<DismissableLayerElement>,
```

OR just use general terms, since it doesn't really matter what the arrays are:

```suggestion
/**
 * Returns the index of the last item that is found in the given subset.
 * Returns -1 if no items are found.
 */
function getHighestIndex(
  orderedItems: unknown[],
  itemsToLookFor: Set<unknown>,
```

Or a middle ground 😅

```suggestion
/**
 * Returns the index of the last layer that is found in the given subset.
 * Returns -1 if no layers are found.
 */
function findHighestLayerIndex(
  orderedLayers: DismissableLayerElement[],
  layersToLookFor: Set<DismissableLayerElement>,
```